packages/flutter/lib/src/painting/text_span.dart

```diff
@@ -279,7 +279,7 @@ class TextSpan extends InlineSpan {
   @override
   void computeSemanticsInformation(List<InlineSpanSemanticsInformation> collector) {
     assert(debugAssertIsValid());
-    if (text != null || semanticsLabel != null) {
+    if (text != null) {
```

Comment:
I'm wondering what real use cases would cause this to trigger, and if we should support them better by doing `text ?? semanticsLabel` on L284 below.

Reply:
I don't know of a use case for a `TextSpan` with a semantics label but no text.

`InlineSpanSemanticsInformation` specifically requires that `text` be non-null, so this logic should be adjusted to match.

Reply:
This seems like an error. If someone expected a screen reader to read the label it wouldn't happen if the text was null. I think we should add an assert that it's not the case and fix the other issue in the linked bug about why the user tried this